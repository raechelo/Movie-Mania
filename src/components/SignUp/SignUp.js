import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';

class SignUp extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      email: '',
      password: '',
      status: ''
    }
  }

  handleSubmit = (e) => {
    e.preventDefault()
    this.createUser(this.state);
  }

  createUser = (state) => {
    const {name, email, password} = state
    const url = 'http://localhost:3000/api/users/new'
    const options = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        name,
        email,
        password
      })
    }
    fetch(url, options)
    .then(response => this.setState({status: response.status}))
    .catch(error => console.log('error', error));
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <h2>Create Account</h2>
        <label htmlFor="name">Name</label>
        <input onChange={ this.handleChange } name='name' type='text' id="name" className='userProp' />
        <label htmlFor="email">Email</label>
        <input onChange={ this.handleChange } name='email' type='email' id="email" className='userProp' />
        <label htmlFor="password">Password</label>
        <input onChange={ this.handleChange } name='password' type='password' id="password" className='userProp' />
        <button>Create</button>
        {this.state.status === 200 && <Redirect to='/' />}
      </form>
    ) 
  }
}

export default SignUp;