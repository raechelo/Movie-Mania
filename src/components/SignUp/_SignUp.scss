.user-form {
  background-color: rgba(161, 161, 161, 0.8);
  box-shadow: 1px 1px 2px rgb(20, 20, 20);
  color: white;
  font-family: $bodyfont;
  font-size: 1.3rem;
  margin: auto auto;
  padding: 1rem 3rem;
  text-align: center;

  input {
    display: block;
    font-family: $bodyfont;
    font-size: 1rem;
    padding: 2px 5px;
    margin: auto;

    &:focus {
      border: 2px solid $accent;
      outline: none;
    }
  }

  label {
    font-size: 1.2rem;
  }

  button {
    border: 2px solid $accent;
    margin: 1rem;
    font-size: 1rem;
    width: 60%;
    &:hover {
      background-image: linear-gradient(to left, $accent 50%, white 50%);
      background-size: 200% 100%;
      background-position: right bottom;
      transition: all .8s ease-out;
      color: white;
    }
  }
}