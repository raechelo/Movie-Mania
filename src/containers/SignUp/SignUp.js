 import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { logInUser } from '../../actions';
import { createNewUser } from '../../thunks/createNewUser';

export class SignUp extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      email: '',
      password: '',
      status: ''
    }
  }

  handleSubmit = async (e) => {
    e.preventDefault()
    const response = await this.props.createNewUser(this.state);
    response ? this.setState({status: response.status}) : this.showError();
  }

  createUser = (state) => {
    const {name, email, password} = state
    const user = {name, email, password}
    const url = 'http://localhost:3000/api/users/new'
    const options = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        name,
        email,
        password
      })
    }
    fetch(url, options)
    .then(response => {
      console.log('response ok?:', response.ok)
      console.log('response:', response)
      this.setState({status: response.status})
      response.ok ? this.props.logInUser(user) : this.showError()
    })
    .catch(error => console.log('error', error));
  }

  showError = () => {
    const errorText = document.querySelector('.error');
    errorText.style.color = '#760a0a';
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  render() {
    return (
      <form className="user-form" onSubmit={this.handleSubmit}>
        <h2>Create Account</h2>
        <p className="error">Email has already been used</p>
        <label htmlFor="name">Name</label>
        <input onChange={ this.handleChange } name='name' type='text' id="name" className='userProp' />
        <label htmlFor="email">Email</label>
        <input onChange={ this.handleChange } name='email' type='email' id="email" className='userProp' />
        <label htmlFor="password">Password</label>
        <input onChange={ this.handleChange } name='password' type='password' id="password" className='userProp' />
        <button>Create <i className="fas fa-angle-right"></i></button>
        {this.state.status === 'success' && <Redirect to='/' />}
      </form>
    ) 
  }
}

SignUp.propTypes = {
  logInUser: PropTypes.func
}

export const mapDispatchToProps = dispatch => {
  return ({
    logInUser: (user) => dispatch(logInUser(user)),
    createNewUser: (user) => dispatch(createNewUser(user))
  })
}

export default connect(null, mapDispatchToProps)(SignUp);