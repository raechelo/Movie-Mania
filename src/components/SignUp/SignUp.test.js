import React from 'react';
import { shallow } from 'enzyme';
import { SignUp } from './SignUp';

describe('SignUp', () => {

  let wrapper;

  it('should match the snapshot with all the data passed in', () => {
    wrapper = shallow( <SignUp /> )
    expect(wrapper).toMatchSnapshot()
  })
})