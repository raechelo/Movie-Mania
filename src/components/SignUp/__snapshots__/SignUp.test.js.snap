// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should match the snapshot with all the data passed in 1`] = `
<form
  className="user-form"
  onSubmit={[Function]}
>
  <h2>
    Create Account
  </h2>
  <p
    class="error"
  >
    Email has already been used
  </p>
  <label
    htmlFor="name"
  >
    Name
  </label>
  <input
    className="userProp"
    id="name"
    name="name"
    onChange={[Function]}
    type="text"
  />
  <label
    htmlFor="email"
  >
    Email
  </label>
  <input
    className="userProp"
    id="email"
    name="email"
    onChange={[Function]}
    type="email"
  />
  <label
    htmlFor="password"
  >
    Password
  </label>
  <input
    className="userProp"
    id="password"
    name="password"
    onChange={[Function]}
    type="password"
  />
  <button>
    Create
  </button>
</form>
`;
